lib/devise/models/database_authenticatable.rb

```diff
@@ -60,17 +60,18 @@ def self.required_fields(klass)
       # the hashed password.
       def password=(new_password)
         @password = new_password
-        self.encrypted_password = password_digest(@password) 
+        self.encrypted_password = password_digest(@password) if @password.present?
       end
 
       # Verifies whether a password (ie from sign in) is the user password.
       def valid_password?(password)
+        return false if password.blank?
```

Comment:
I guess this line doesn't need to be here, right? It was added in https://github.com/plataformatec/devise/pull/4261/files#diff-19b4dd928714c72f1338874351e8ff2dR45

Reply:
Fixed by doing the reverts on chronological order